Makefile

```diff
@@ -0,0 +1,39 @@
+# Makefile for running typical developer workflow actions.
+# To run actions in a subdirectory of the repo:
+#   make lint build dir=translate/snippets
+
+INTERFACE_ACTIONS="build test lint"
+
+.ONESHELL: #ease subdirectory work by using the same subshell for all commands
+.-PHONY: *
+
+# Default to current dir if not specified.
+dir ?= $(shell pwd)
+
+export GOOGLE_CLOUD_PROJECT = ${GOOGLE_SAMPLES_PROJECT}
+
+build:
+	cd ${dir}
+	mvn compile
+
+test: check-env build
+	cd ${dir}
+	mvn --quiet --batch-mode --fail-at-end clean verify \
+    -Dfile.encoding="UTF-8" \
+    -Dorg.slf4j.simpleLogger.log.org.apache.maven.cli.transfer.Slf4jMavenTransferListener=warn \
+    -Dmaven.test.redirectTestOutputToFile=true \
+    -Dbigtable.projectID="${GOOGLE_CLOUD_PROJECT}" \
+    -Dbigtable.instanceID=instance
```

Comment:
Both these lines can be removed for local testing.

Reply:
this command comes from the one currently used in the CI testing.
I'd like to eventually move CI to use the Makefile, so i'd prefer to leave them to avoid issues later. Unless there's a reason these options might interfere with local testing?

Reply:
There's no impact to the samples except for testing bigtable. The suggestion was to keep the Makefile tightly coupled to local testing if that's what you are targeting now.

I suggest the two options:
1. We can either remove these and add back when the Makefile is being made CI compatible
2. Or, init both these params to empty values to avoid any resource mixups.

Reply:
it looks like most of the bigtable tests use [`BIGTABLE_TESTING_INSTANCE`](https://github.com/GoogleCloudPlatform/java-docs-samples/blob/d3170e9b64520e667475480494c9353b7a28fa60/bigtable/hbase/snippets/src/test/java/com/example/bigtable/ConfigureConnectionPoolTest.java#L46).

these properties appear to only be used in [a single test](https://github.com/GoogleCloudPlatform/java-docs-samples/blob/d3170e9b64520e667475480494c9353b7a28fa60/bigtable/hbase/snippets/src/test/java/com/example/bigtable/BigtableConnectTest.java#L34).

I'm inclined to leave these in place for now, to avoid breaking even the single test that depends on them. We could also file an issue to update the single test to use the environment variable instead, which would let us safely remove these properties from both this invocation, and the CI script. WDYT?